lib/situation_room/check/get_request.ex

```diff
@@ -0,0 +1,32 @@
+defmodule SituationRoom.Check.GetRequest do
+  @moduledoc """
+  Module that sends a head request to a site's url
+  Retrieves a status code and a response time
+  """
+  alias SituationRoom.Site.Check
+
+  def head(%SituationRoom.Site{} = site) do
+    start = :os.system_time(:microsecond)
+    response = SituationRoom.Check.Client.head(site.endpoint)
+    time = (:os.system_time(:microsecond) - start) / 1000
+
+    case response do
+      {:ok, %Tesla.Env{} = env} ->
+        %Check{
```

Comment:
These check structs don't need to be the the schema structs. You could do a defstruct somewhere and just use it as a DTO. Then each check could insert any information it wanted (by passing both to `run`), or you could just create the ecto struct somewhere else and not let the checks know about it. 

The other option if we do want to pass around the ecto structs is you can make `virtual` fields on them so they can hold information that doesn't get persisted to the database.